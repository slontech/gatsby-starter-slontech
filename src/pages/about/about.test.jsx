import React from 'react';
import { shallow } from 'enzyme';

import About from './index';

describe('About', () => {
  const Component = shallow(<About />);
  it('renders 1 component', () => {
    expect(Component).toHaveLength(1);
  });

  it('is of type <div/>', () => {
    expect(Component.type()).toBe('div');
  });

  it('matches snapshot', () => {
    const aboutPage = shallow(<About />);
    expect(aboutPage).toMatchSnapshot();
  });
});
