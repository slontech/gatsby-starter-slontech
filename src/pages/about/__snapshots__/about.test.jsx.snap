// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AboutPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Hi from the About page
        </h1>,
        <p>
          nothing to see here
        </p>,
        <GatsbyLink
          to="/"
        >
          Go back to the homepage
        </GatsbyLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hi from the About page",
        },
        "ref": null,
        "rendered": "Hi from the About page",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "nothing to see here",
        },
        "ref": null,
        "rendered": "nothing to see here",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Go back to the homepage",
          "to": "/",
        },
        "ref": null,
        "rendered": "Go back to the homepage",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Hi from the About page
          </h1>,
          <p>
            nothing to see here
          </p>,
          <GatsbyLink
            to="/"
          >
            Go back to the homepage
          </GatsbyLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hi from the About page",
          },
          "ref": null,
          "rendered": "Hi from the About page",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "nothing to see here",
          },
          "ref": null,
          "rendered": "nothing to see here",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Go back to the homepage",
            "to": "/",
          },
          "ref": null,
          "rendered": "Go back to the homepage",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
